### PR TITLE
Dotnet user agent

### PR DESCRIPTION
**Pull request by cole-bq**
No integration tests implemented yet. Manually testing confirms server is receiving user agent when expected.

### CODE REVIEW

File: gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPoolSettings.cs [hunk 1]
```diff
@@ -131,5 +133,19 @@ public TimeSpan ReconnectionBaseDelay
...
+        /// <remarks>
+        /// The default value is true. When enabled, user agents will only be sent during the web socket
+        /// handshake. User agent follows the form:
+        /// [Application Name] [GLV Name].[Version] [Language Runtime Version] [OS].[Version] [CPU Architecture]
+        /// </remarks>
+        public bool EnableUserAgentOnConnect
+        {
+            get => _enableUserAgentOnConnect;
+            set => _enableUserAgentOnConnect = value;
+        }
```
Comment (vkagamlyk) [inline gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPoolSettings.cs:149]: No need to introduce field
```suggestion
        public bool EnableUserAgentOnConnect  { get;  set; } = DefaultEnableUserAgentOnConnect;
```

File: gremlin-dotnet/src/Gremlin.Net/Driver/GremlinClient.cs [hunk 2]
```diff
@@ -108,7 +108,11 @@ public class GremlinClient : IGremlinClient
 
             var connectionFactory =
                 new ConnectionFactory(gremlinServer, messageSerializer,
-                    new WebSocketSettings { WebSocketConfigurationCallback = webSocketConfiguration }, sessionId);
+                    new WebSocketSettings
+                    {
+                        WebSocketConfigurationCallback = webSocketConfiguration,
+                        EnableUserAgentOnConnect = connectionPoolSettings?.EnableUserAgentOnConnect ?? true
```
Comment (vkagamlyk) [inline gremlin-dotnet/src/Gremlin.Net/Driver/GremlinClient.cs:114]: there are `DefaultEnableUserAgentOnConnect` const, better to use it not just `true`

File: gremlin-dotnet/src/Gremlin.Net/Process/Utils.cs [hunk 3]
```diff
@@ -63,5 +64,23 @@ public static void Forget(this Task task)
...
+
+        /// <summary>
+        ///  Returns a user agent for connection request headers.
+        ///
+        /// Format:
+        /// "[Application Name] [GLV Name].[Version] [Language Runtime Version] [OS].[Version] [CPU Architecture]"
+        /// </summary>
+        public static string GetUserAgent()
+        {
+            string applicationName = Assembly.GetExecutingAssembly().GetName().Name.Replace(' ', '_');
```
Comment (vkagamlyk) [inline gremlin-dotnet/src/Gremlin.Net/Process/Utils.cs:76]: project convention is to use `var` keyword for declarations

```suggestion
            var applicationName = Assembly.GetExecutingAssembly().GetName().Name.Replace(' ', '_');
```

File: gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPoolSettings.cs (hunk 1 above)
Comment (cole-bq) [reply gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPoolSettings.cs]: Thanks that looks a lot cleaner

File: gremlin-dotnet/src/Gremlin.Net/Driver/GremlinClient.cs [hunk 4]
```diff
@@ -171,7 +175,8 @@ public class GremlinClient : IGremlinClient
             messageSerializer ??= new GraphSON3MessageSerializer();
             var webSocketSettings = new WebSocketSettings
             {
-                WebSocketConfigurationCallback = webSocketConfiguration
+                WebSocketConfigurationCallback = webSocketConfiguration,
+                EnableUserAgentOnConnect = connectionPoolSettings?.EnableUserAgentOnConnect ?? true
```
Comment (vkagamlyk) [inline gremlin-dotnet/src/Gremlin.Net/Driver/GremlinClient.cs:179]:
```suggestion
                EnableUserAgentOnConnect = connectionPoolSettings?.EnableUserAgentOnConnect ?? ConnectionPoolSettings.DefaultEnableUserAgentOnConnect
```

**Pull request closed without merging**